exercises/practice/parallel-letter-frequency/parallel_letter_frequency_test.rb

```diff
@@ -0,0 +1,212 @@
+    expected = {}
+    actual = ParallelLetterFrequency.count(texts)
+    assert_equal expected, actual
+  end
+
+  def test_ignore_punctuation
+    skip
+    texts = ['!', '?', ';', ',', '.', '-', "'", '"', '/', ':', '{}', '[]', '()']
+    expected = {}
+    actual = ParallelLetterFrequency.count(texts)
+    assert_equal expected, actual
+  end
+
+  def test_ignore_numbers
+    skip
+    texts = %w[1 2 3 4 5 6 7 8 9 0]
+    expected = {}
+    actual = ParallelLetterFrequency.count(texts)
+    assert_equal expected, actual
+  end
+
+  def test_unicode_letters
+    skip
+    texts = %w[本 φ ほ ø]
+    expected = { '本' => 1, 'φ' => 1, 'ほ' => 1, 'ø' => 1 }
+    actual = ParallelLetterFrequency.count(texts)
+    assert_equal expected, actual
+  end
+
+  def test_combo_lower_upper_punctuation_whitespace
+    skip
+
+    texts = [ruby_wiki]
+    expected = {
+      "r" => 34,
+      "u" => 20,
+      "b" => 10,
+      "y" => 14,
+      "i" => 53,
+      "s" => 26,
+      "a" => 46,
+      "n" => 37,
+      "t" => 34,
+      "e" => 39,
+      "p" => 28,
+      "d" => 24,
+      "h" => 10,
+      "g" => 23,
+      "l" => 26,
+      "v" => 6,
+      "o" => 26,
+      "m" => 25,
+      "w" => 5,
+      "c" => 18,
+      "j" => 5,
+      "k" => 2,
+      "z" => 1,
+      "f" => 4
+    }
+    actual = ParallelLetterFrequency.count(texts)
+    assert_equal expected, actual
+  end
+
+  def test_large_texts
+    skip
+    texts = [ch37, ch38, ch39]
+    expected = {
+      "f" => 89,
+      "o" => 333,
+      "r" => 250,
+      "e" => 570,
+      "t" => 429,
+      "p" => 57,
+      "s" => 311,
+      "u" => 131,
+      "b" => 110,
+      "l" => 260,
+      "a" => 356,
+      "n" => 278,
+      "d" => 184,
+      "m" => 156,
+      "i" => 352,
+      "g" => 120,
+      "c" => 77,
+      "h" => 308,
+      "y" => 113,
+      "v" => 52,
+      "k" => 49,
+      "q" => 2,
+      "w" => 133,
+      "x" => 2,
+      "j" => 6,
+      "z" => 4
+    }
+    actual = ParallelLetterFrequency.count(texts)
+    assert_equal expected, actual
+  end
+
+  def test_many_small_texts
+    skip
+    texts = Array.new(100, 'abc')
+    actual = ParallelLetterFrequency.count(texts)
+    expected = { 'a' => 100, 'b' => 100, 'c' => 100 }
+    assert_equal expected, actual
+  end
+
+  def ruby_wiki
+    <<~TEXT
+          Ruby is an interpreted, high-level, general-purpose programming language which supports multiple programming paradigms. It was designed with an emphasis on programming productivity and simplicity. In Ruby, everything is an object, including primitive data types. It was developed in the mid-1990s by Yukihiro "Matz" Matsumoto in Japan.
+
+      Ruby is dynamically typed and uses garbage collection and just-in-time compilation. It supports multiple programming paradigms, including procedural, object-oriented, and functional programming. According to the creator, Ruby was influenced by Perl, Smalltalk, Eiffel, Ada, BASIC, Java, and Lisp.
+    TEXT
```

Comment:
The reason for the `~` in the heredoc is to be able to trim the left "extra" whitespace.  This means that we can indent the text so that is "nice" for the source code.

But if we are not using the `~` feature, then we should not use that Heredoc style.

Reply:
Moved to separate text files.